src/CodeGen_C.cpp

```diff
@@ -1375,14 +1375,14 @@ string CodeGen_C::print_type(Type type, AppendSpaceIfNeeded space_option) {
 string CodeGen_C::print_reinterpret(Type type, const Expr &e) {
     ostringstream oss;
     if (type.is_handle() || e.type().is_handle()) {
-        // Use a c-style cast if either src or dest is a handle --
+        // Use a reinterpret_cast<> if either src or dest is a handle --
         // note that although Halide declares a "Handle" to always be 64 bits,
         // the source "handle" might actually be a 32-bit pointer (from
-        // a function parameter), so calling reinterpret<> (which just memcpy's)
+        // a function parameter), so calling halide_reinterpret_bits<> (which just memcpy's)
         // would be garbage-producing.
-        oss << "(" << print_type(type) << ")";
+        oss << "reinterpret_cast<" << print_type(type) << ">";
```

Comment:
I don't think a reinterpret cast is valid here, because the bit-width might change.